hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyInputStream.java

```diff
@@ -96,12 +114,13 @@ public static LengthInputStream getFromOmKeyInfo(OmKeyInfo keyInfo,
     List<OmKeyLocationInfo> keyLocationInfos = keyInfo
```

Comment:
```suggestion
public static LengthInputStream getFromOmKeyInfo(OmKeyInfo keyInfo, List<OmKeyLocationInfo> keyLocationInfos,
      XceiverClientFactory xceiverClientFactory,
      boolean verifyChecksum,  Function<OmKeyInfo, OmKeyInfo> retryFunction,
      BlockInputStreamFactory blockStreamFactory) {
      List<BlockExtendedInputStream> streams = createStreams(keyInfo,
        keyLocationInfos, xceiverClientFactory, verifyChecksum, retryFunction,
        blockStreamFactory);
    KeyInputStream keyInputStream =
        new KeyInputStream(keyInfo.getKeyName(), streams);

    return new LengthInputStream(keyInputStream, keyInputStream.getLength());
      
      }
      
      public static LengthInputStream getFromOmKeyInfo(OmKeyInfo keyInfo,
      XceiverClientFactory xceiverClientFactory,
      boolean verifyChecksum,  Function<OmKeyInfo, OmKeyInfo> retryFunction,
      BlockInputStreamFactory blockStreamFactory) {
       return getFromOmKeyInfo(keyInfo,  keyInfo
        .getLatestVersionLocations().getBlocksLatestVersionOnly(), xceiverClientFactory, retryFunction, blockStreamFactory)
      }
```
Adding this could help you reuse a lot of code for getStreamsFromKeyInfo() function

Reply:
You could make the new function with private instead. Forgot to change it while copying it

Reply:
Done